src/types/synchronization.d.ts

```diff
@@ -21,15 +21,27 @@ export interface MetadataPackage {
 export interface SynchronizationResult extends MetadataImportResponse {
     instance: {
         id: string;
-        name: string;
-        url: string;
+        name?: string;
+        url?: string;
     };
-    report: {
-        typeStats: any[];
+    report?: {
+        typeStats: MetadataImportStats[];
         messages: any[];
     };
 }
 
+export type SynchronizationReportStatus = "READY" | "RUNNING" | "FAILURE" | "DONE";
+
+export interface SynchronizationReport {
+    id?: string;
```

Comment:
I guess the id is left optional because there is no id when we use build. Not necessary here, just as a note, I looked at how to have types without some fields. build would use the id-less type. Example:

```
type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

type Pos3D = { x: number; y: number; z: number };
type Pos2D = Omit<Pos3D, "z">;
type Pos1D = Omit<Pos3D, "y" | "z">;
```